ui/components-react/src/components-react/filter-builder/FilterBuilderState.ts

```diff
@@ -117,10 +118,10 @@ export function isPropertyFilterBuilderRuleGroup(item: PropertyFilterBuilderRule
 }
 
 /** @alpha */
-export function usePropertyFilterBuilderState() {
-  const [state, setState] = React.useState<PropertyFilterBuilderState>(() => ({
-    rootGroup: createEmptyRuleGroup(),
-  }));
+export function usePropertyFilterBuilderState(initialFilter?: PropertyFilter) {
+  const [state, setState] = React.useState<PropertyFilterBuilderState>(
+    () => initialFilter ? convertFilterToState(initialFilter) : { rootGroup: createEmptyRuleGroup() }
+  );
```

Comment:
If the initial filter is modified after the first render, this construct will not take it into account (function will not be rerun), this will cause problem with the upcoming React18. (which may do the initial render twice)

Would it be correct to have a `useEffect` that resets the state if the initialFilter changes ?

Reply:
My idea was to only use `initialFilter` from first render and don't react to `initialFilter` changes as this is an uncontrolled component and state should be modified only by user interactions.

Reply:
@raplemie

Reply:
Sorry, I thought I resolved that last week.